src/bootstrap/test.rs

```diff
@@ -1057,11 +1057,9 @@ impl Step for Compiletest {
         builder.ensure(compile::Test { compiler, target: compiler.host });
         // Also provide `rust_test_helpers` for the host.
         builder.ensure(native::TestHelpers { target: compiler.host });
+        // As well as the target
+        builder.ensure(native::TestHelpers { target });
 
-        // wasm32 can't build the test helpers
-        if !target.contains("wasm32") {
-            builder.ensure(native::TestHelpers { target });
-        }
```

Comment:
I'm not sure which LLVM versions rust supports, so this may not be correct. This change was necessary to uncover all the ABI problems, though.

Reply:
FWIW this *may* fail on CI for the wasm32-unknown-unknown target, but I think that it'll probably work in that it'll generate a native library with ELF code (using `clang`) and then we'd just ignore it during the test suite. In that sense let's just ignore this for now and if it causes problems on CI we can add it back, but only for the non-emscripten target

Reply:
I believe this is the line that's failing on CI, this'll need to be disabled still for wasm32-unknown-unknown I believe